contrib/gorilla/mux/mux_test.go

```diff
@@ -182,6 +182,25 @@ func TestAnalyticsSettings(t *testing.T) {
 	})
 }
 
+func TestResourceNamer(t *testing.T) {
+	staticNamer := func(*Router, *http.Request) string {
+		return "static resource name"
+	}
+
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+	mux := NewRouter(WithResourceNamer(staticNamer))
+	mux.Handle("/200", okHandler()).Host("localhost")
+	r := httptest.NewRequest("GET", "http://localhost/200", nil)
+	w := httptest.NewRecorder()
+	mux.ServeHTTP(w, r)
+
+	spans := mt.FinishedSpans()
+	assert.Equal(1, len(spans))
+	assert.Equal("GET static resource name", spans[0].Tag(ext.ResourceName))
```

Comment:
We want this to work just like our other integrations ([example](https://github.com/DataDog/dd-trace-go/blob/876b6b3508944c24f204f96d93474b3ec5dd72c6/contrib/gin-gonic/gin/gintrace_test.go#L333)) for consistency and have it include the request method.

What I'm trying to say is that when using `WithResourceNamer` it should overwrite the entire resource including the method.

Reply:
Yeah I think I looked at the elastic integration as inspiration which only returned the path-part of the resource name but I agree it would be better to be able to set the whole thing. I've adjusted accordingly.